src/targets/gpu/device/where.cpp

```diff
@@ -0,0 +1,30 @@
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/gpu/device/where.hpp>
+#include <migraphx/gpu/device/tensor.hpp>
+#include <migraphx/gpu/device/launch.hpp>
+#include <migraphx/gpu/device/types.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+argument where(hipStream_t stream, const argument& arg0, const argument& arg1, const argument& arg2)
+{
+    auto s = arg1.get_shape();
+    argument result{s};
```

Comment:
This should take the output buffer as input, and this should use `nary` function(similiar to the other pointtwise operators we do).